Bluetooth: Mesh: Add Light LC Server resume timer (#4634)
nrfconnect/sdk-nrf: include/bluetooth/mesh/light_ctrl_srv.rst

.. note::
    The state machine only works while the Light LC Server is enabled, and it always starts in the Standby state.

Resuming state machine operation
--------------------------------

Whenever something other than the Light LC Server interacts the controlled Lightness Server, the Light LC Server will disable its state machine, and the Lightness Server starts running independently.
To resume state machine operation, the Light LC Server must explicitly be re-enabled.

To avoid being stuck with a Lightness Server that runs independently forever, the Light LC Server implements a *Resume timer*, that lets the Light LC Server regain control after being disabled for a certain number of seconds.
The Resume timer can be configured with the :option:`CONFIG_BT_MESH_LIGHT_CTRL_SRV_RESUME_DELAY` option, and is disabled by default.

.. note::
    The resume timer does not exist in the Bluetooth Mesh specification, and may become incompatible with future specification changes.

>     Although it does not break the specification or qualification tests in the current iteration of the Bluetooth Mesh Specification, its behavior may be unexpected for third party devices, and should be used with caution.

Author: @mia-ko input?

>     Although it does not break the specification or qualification tests in the current iteration of the Bluetooth Mesh Specification, its behavior may be unexpected for third party devices, and should be used with caution.

Reviewer: It should be Bluetooth mesh specification :)
And I think we should stick to resume timer too, i.e. not Resume timer.

> Resuming state machine operation
> --------------------------------
> 
> Whenever something other than the Light LC Server interacts the controlled Lightness Server, the Light LC Server will disable its state machine, and the Lightness Server starts running independently.
> To resume state machine operation, the Light LC Server must explicitly be re-enabled.
> 
> To avoid being stuck with a Lightness Server that runs independently forever, the Light LC Server implements a *Resume timer*, that lets the Light LC Server regain control after being disabled for a certain number of seconds.
> The Resume timer can be configured with the :option:`CONFIG_BT_MESH_LIGHT_CTRL_SRV_RESUME_DELAY` option, and is disabled by default.
> 
> .. note::
>     The resume timer does not exist in the Bluetooth Mesh specification, and may become incompatible with future specification changes.
>     Although it does not break the specification or qualification tests in the current iteration of the Bluetooth Mesh Specification, its behavior may be unexpected for third party devices, and should be used with caution.

Reviewer: ```suggestion
Resuming the state machine operation
------------------------------------

Whenever something but the Light LC Server interacts with the controlled Lightness Server, the Light LC Server disables its state machine, and the Lightness Server starts running independently.
To resume the state machine operation, the Light LC Server must be explicitly re-enabled.

To avoid having a Lightness Server running independently forever, the Light LC Server implements a resume timer that lets the Light LC Server regain control after being disabled for a certain number of seconds.
The resume timer can be configured with the :option:`CONFIG_BT_MESH_LIGHT_CTRL_SRV_RESUME_DELAY` option, and is disabled by default.

.. note::
    The resume timer does not exist in the Bluetooth mesh specification, and may become incompatible with future specification changes.
    Although it does not break the specification or qualification tests in the current iteration of the Bluetooth mesh specification, its behavior may be unexpected for third party devices, and should be used with caution.
```
